src/v3/src/util/settingsUtils.ts

```diff
@@ -246,3 +248,33 @@ export const isOauth2Enabled = (widgetProps: WidgetProps): boolean => {
   }
   return typeof clientId !== 'undefined' && authScheme?.toLowerCase() === 'oauth2';
 };
+
+export const getPageTitle = (
+  widgetProps: WidgetProps,
+  formTitle: string,
+  idxTransaction?: IdxTransaction,
+): string | null => {
+  const { brandName, features: { setPageTitle } = {} } = widgetProps;
+  const eventContext: EventContext = typeof idxTransaction === 'undefined'
+    ? { controller: null }
+    : getEventContext(idxTransaction);
+
+  switch (typeof setPageTitle) {
+    // When setPageTitle option is 'undefined', default will be to set title based on page header
+    // When setPageTitle option is 'true', set title based on page header
+    case 'boolean':
+    case 'undefined':
+      if (setPageTitle === false) {
+        // Indicates setPageTitle config option was purposefully disabled
+        return null;
+      }
+      return brandName ? `${brandName} | ${formTitle}` : formTitle;
+    case 'string':
+      return setPageTitle;
+    case 'function':
+      return setPageTitle(eventContext, { formTitle, brandName });
```

Comment:
Maybe this could be passed a callack or the default implementation output?

Reply:
Not sure if this will suffice, but, in the callback, we provide the default page title as a param (to allow the customer to return it from the function or modify it). I tried to demonstrate that [here](https://github.com/okta/okta-signin-widget/pull/3331/files#diff-1792a790ad357d74f93cb579423320ab3d01d42a677ab31eca7b28b0555ba817R186-R222) in this test. 

Let me know if this is not good enough and I can add a more dedicated test for this.